lib/workers/repository/config-migration/branch/commit-message.ts

```diff
@@ -1,20 +1,23 @@
 import type { RenovateConfig } from '../../../../config/types';
+import { clone } from '../../../../util/clone';
 import * as template from '../../../../util/template';
 import type { CommitMessage } from '../../model/commit-message';
 import { CommitMessageFactory } from '../../model/commit-message-factory';
 
-export class ConfigMigrationCommitMessageFactory {
+export class ConfigMigrationSemanticFactory {
   private readonly config: RenovateConfig;
+  private commitMessage: string | null = null;
+  private prTitle: string | null = null;
 
-  private readonly configFile: string;
-
-  constructor(config: RenovateConfig, configFile: string) {
-    this.config = config;
-    this.configFile = configFile;
+  constructor(config: RenovateConfig, private readonly configFile: string) {
+    this.config = clone(config);
   }
 
-  create(): CommitMessage {
+  private create(isTitle = false): CommitMessage {
     const { commitMessage } = this.config;
+    const commitMessageTopic = isTitle
```

Comment:
i think this will close and open a lot PR

Reply:
we currently searching migration PRs only by its branch name, which hasnt changed, so i dont think we are going to open/close prs because of this change.

note we recently had another PR title change in https://github.com/renovatebot/renovate/pull/16490

Reply:
Will we *rename* PRs we find? I don't think we were doing that before, and maybe it's for a separate PR?

Reply:
currently we won't rename PRs, ill open another separate one for it.

Reply:
when we do this later, renovate has already closed the PR, so we need the check in this PR.

Reply:
i might not follow, my approach for renaming PRs would be to only check open PRs we find, and update the PR if current title is not what is expected.

Reply:
we are not closing any PRs because of title miss match if that what you meant